File: Single-Linked-List/traverse_single_linked_list.py
# File Name: traverse_list.py
# Program to traverse and print a singly linked list from left to right

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

head = None

# Function to traverse and print the list
def traverse():
    global head

    if head is None:
        print("List is empty.")
        return

    temp = head
    print("Linked List contents:", end=" ")
    while temp is not None:
        print(temp.data, end=" -> ")
        temp = temp.next
    print("NULL")

if __name__ == "__main__":
    traverse()

File: Single-Linked-List/TraverseSingleLinkedList.java
// File Name: TraverseList.java
// Program to traverse and print a singly linked list from left to right

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class TraverseList {
    static Node head = null;

    // Function to traverse and print the list
    static void traverse() {
        if (head == null) {
            System.out.println("List is empty.");
            return;
        }

        Node temp = head;
        System.out.print("Linked List contents: ");
        while (temp != null) {
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.println("NULL");
    }

    public static void main(String[] args) {
        traverse();
    }
}
